src/main/java/somethingrandom/entity/ActionableItem.java

```diff
@@ -13,55 +13,43 @@ public class ActionableItem extends Item {
     /**
      * The Name.
      */
-    public String name;
+    private String name;
     /**
      * The ID.
      */
-    public final UUID id;
+    private final UUID id;
     /**
      * The Creation date.
      */
 
-    public final Instant creationDate;
+    private final Instant creationDate;
     /**
      * The amount of time needed to complete the task.
      */
     @Nullable
-    public Duration neededTime;
-    /**
-     * The Item kind.
-     */
-    public final ItemKind itemKind;
+    private Duration neededTime;
 
     /**
      * Instantiates a new Actionable item.
      *
      * @param name         the name
      * @param creationDate the creation date
      * @param neededTime   the needed time
-     * @param itemKind     the item kind
      */
-    public ActionableItem(String name, Instant creationDate, @Nullable Duration neededTime, ItemKind itemKind){
+    public ActionableItem(String name, Instant creationDate, @Nullable Duration neededTime){
         this.name = name;
         this.id = UUID.randomUUID();
         this.creationDate = creationDate;
         this.neededTime = neededTime;
-        this.itemKind = itemKind;
-    }
-    public String getName(){
-        return this.name;
-    }
-    public UUID getID(){
-        return this.id;
     }
-    public String setName(String name){
-        this.name = name;
-        return this.name;
-    }
-    public Instant getCreationDate(){
-        return this.creationDate;
+    /**
+     * Returns item kind.
+     *
+     * @return the item kind
+     */
+    public ItemKind getItemKind(){
```

Comment:
We'll have to discuss what to do with ItemKind. Maybe it'd be worth leaving out for now, and adding back in later if we need it? (It seems like a massive OCP violation, and I think it basically just needs to be input to a factory in the interface adapters; the entities might not need to know their kinds at all.)